main.c

```diff
@@ -40,36 +45,38 @@ void insertNode (int position ,int item, struct node **ptrtohead)
 int main()
 {
     // Create Linked list
-    struct node *head= NULL, *temp, *constanthead;
+    struct node *current = NULL, *temp;
     int n ;
     int i ;
     int item;
-    struct node **ptrtohead = &head;
+    struct node* ptrtohead; /* declare a canvas of art that is blank(null) */
+
     scanf("%d", &n);
     getchar();
-    for ( i=0; i<n; i++)
-    {
+    for ( i=0; i<n; i++) {
+	/* C is a sequential language, so we scan user input first
+	 *  before deciding what to do with the values
+	 */
+        scanf("%d", &item);
+        getchar();
 
-        if (i==0){
-            head = malloc(sizeof(struct node));
-            constanthead = head;
-        }
-        else{
+        if (i==0) {
+            current = malloc(sizeof(struct node));
+	    current->item = item;
+	    ptrtohead = current;
+        } else {
             temp = malloc(sizeof(struct node));
-            head->nxt= temp;
-            head = temp;
+	    temp->item = item;
+            current->nxt= temp;
+            current = temp;
         }

```

Comment:
This should be fairly self-explanatory, but do let me know if you need a further explanation.